Create SQLCode.sql

Jonasgith opened the pull request.

Jonasgith committed:
Create SQLCode.sql

supermar1010 commented on sql/SQLCode.sql:
```diff
@@ -0,0 +1,26 @@
+CREATE User(
+user_ID VARCHAR(5),
```
```suggestion
user_id INT AUTO_INCREMENT PRIMARY KEY,
```

supermar1010 replied:
This should be applied to every id. And please also change the foreignkeys datatypes to match

supermar1010 commented on sql/SQLCode.sql:
```diff
@@ -0,0 +1,26 @@
+u_name VARCHAR(30) NOT NULL,
+u_pwd VARCHAR(30) NOT NULL,
+is_admin BIT DEFAULT 0,
+PRIMARY KEY (user_ID)
```
```suggestion

```

supermar1010 replied:
Same for the other tables, this works but I think its overly verbose when you can also use the short syntax.
As shown above

supermar1010 commented on sql/SQLCode.sql:
```diff
@@ -0,0 +1,26 @@
+CREATE MetaData(
+ID VARCHAR(5),
+file_size DECIMAL(4,0) NOT NULL,
+date_added DATE,
```
I suggest we use int here, because then we can just use the unix timestamp which is just the number of seconds since 1970.
Then we don't have to crazy parsing and stuff to get a proper datetime

```suggestion
date_added INT,
```

supermar1010 commented on sql/SQLCode.sql:
```diff
@@ -0,0 +1,26 @@
+);
+
+CREATE File(
+file_ID VARCHAR(5),
```
I guess its implicitely clear that this is the id of a file, because it is in the table file, so in my opinion we can remove these prefixes
```suggestion
id VARCHAR(5),
```

Jonasgith committed:
Update SQLCode.sql

Updated the -suggested primary key changes
                     -date data type
                     -names of the ID's in User and File

supermar1010 commented on sql/SQLCode.sql:
```diff
@@ -0,0 +1,23 @@
+CREATE User(
+ID INT AUTO_INCREMENT PRIMARY KEY,
+u_name VARCHAR(30) NOT NULL,
```
```suggestion
name VARCHAR(30) NOT NULL,
```

supermar1010 commented on sql/SQLCode.sql:
```diff
@@ -0,0 +1,23 @@
+CREATE User(
+ID INT AUTO_INCREMENT PRIMARY KEY,
+u_name VARCHAR(30) NOT NULL,
+u_pwd VARCHAR(30) NOT NULL,
```
```suggestion
pwd VARCHAR(30) NOT NULL,
```

supermar1010 commented on sql/SQLCode.sql:
```diff
@@ -0,0 +1,23 @@
+CREATE File(
+ID INT AUTO_INCREMENT PRIMARY KEY,
+file_content VARCHAR(30),
+file_location VARCHAR(30),
```
I think the paths can get pretty long

```suggestion
file_location VARCHAR(100),
```

supermar1010 commented on sql/SQLCode.sql:
```diff
@@ -0,0 +1,23 @@
+
+CREATE File(
+ID INT AUTO_INCREMENT PRIMARY KEY,
+file_content VARCHAR(30),
```
```suggestion
file_content LONGTEXT,
```

supermar1010 commented on sql/SQLCode.sql:
```diff
@@ -0,0 +1,23 @@
+file_size DECIMAL(4,0) NOT NULL,
+date_added INT,
+number_of_updates SMALLINT, 
+user_ID VARCHAR(5),
```
Has to be the same type as id of the other table I think
```suggestion
user_ID INT,
```

supermar1010 commented on sql/SQLCode.sql:
```diff
@@ -0,0 +1,23 @@
+date_added INT,
+number_of_updates SMALLINT, 
+user_ID VARCHAR(5),
+file_ID VARCHAR(5),
```
Has to be the same type as id of the other table I think
```suggestion
file_ID INT,
```

supermar1010 commented:
Can you please rename the file to initDb.sql. I think that is a better name

supermar1010 committed:
Apply suggestions from code review

Commit:
Changed name to initDb.sql

supermar1010 committed:
Delete SQLCode.sql
